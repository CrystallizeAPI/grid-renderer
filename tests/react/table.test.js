import React from 'react';
import { shallow } from 'enzyme';
import Table from '../../src/react/table';
import { CustomCellComponent, model } from './utils';

const { rows } = model;

describe('Table', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Table rows={rows} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a TableCell component for each cell', () => {
    const wrapper = shallow(<Table rows={rows} />);
    const cells = wrapper.find('TableCell');
    expect(cells).toHaveLength(3);
    expect(cells.first().prop('cell')).toEqual(rows[0].columns[0]);
  });

  describe('when the `cellComponent` prop is given', () => {
    it('renders the custom component for each cell', () => {
      const wrapper = shallow(
        <Table cellComponent={CustomCellComponent} rows={rows} />
      );
      const cells = wrapper.find(CustomCellComponent);
      expect(cells).toHaveLength(3);
      expect(cells.first().prop('cell')).toEqual(rows[0].columns[0]);
    });
  });

  describe('when the `renderCellContent` prop is given', () => {
    it('calls the prop as a function for each cell', () => {
      const renderCellContent = jest.fn(cell => (
        <CustomCellComponent cell={cell} />
      ));
      const wrapper = shallow(
        <Table renderCellContent={renderCellContent} rows={rows} />
      );
      expect(renderCellContent.mock.calls).toHaveLength(3);
      expect(wrapper.find(CustomCellComponent)).toHaveLength(3);
    });
  });
});
