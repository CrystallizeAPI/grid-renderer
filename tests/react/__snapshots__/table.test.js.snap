// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<styled.div>
  <styled.table>
    <tbody>
      <styled.tr
        key="row-0"
      >
        <TableCell
          cell={
            Object {
              "item": Object {
                "name": "Item 1",
              },
              "layout": Object {
                "colspan": 2,
                "rowspan": 1,
              },
            }
          }
          key="cell-0-0"
          totalColSpan={4}
        />
        <TableCell
          cell={
            Object {
              "item": Object {
                "name": "Item 2",
              },
              "layout": Object {
                "colspan": 2,
                "rowspan": 1,
              },
            }
          }
          key="cell-0-1"
          totalColSpan={4}
        />
      </styled.tr>
      <styled.tr
        key="row-1"
      >
        <TableCell
          cell={
            Object {
              "item": Object {
                "name": "Item 3",
              },
              "layout": Object {
                "colspan": 4,
                "rowspan": 1,
              },
            }
          }
          key="cell-1-0"
          totalColSpan={4}
        />
      </styled.tr>
    </tbody>
  </styled.table>
</styled.div>
`;
